src/transform/ir_proto_to_datalog/utils/access_path.cc

```diff
@@ -0,0 +1,96 @@
+#include "access_path.h"
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_cat.h"
+
+#include <string>
+
+namespace raksha::transform::utils {
+
+// Check the components of the two access paths for equality. First, ensure
+// that they have the same number of components, then check component by
+// component.
+bool AccessPath::operator==(const AccessPath &other) const {
```

Comment:
Won't the compiler generated implementation be sufficient?

Reply:
I think default `operator==` is a C++20 feature, and thus not available to us: https://en.cppreference.com/w/cpp/language/default_comparisons .

Reply:
Simplified this by using `std::vector`'s `operator==`.